editor/plugins/material_editor_plugin.cpp

```diff
@@ -299,14 +343,28 @@ MaterialEditor::MaterialEditor() {
 	vb_light->add_child(light_2_switch);
 	light_2_switch->connect(SceneStringName(pressed), callable_mp(this, &MaterialEditor::_on_light_2_switch_pressed));
 
-	if (EditorSettings::get_singleton()->get_project_metadata("inspector_options", "material_preview_on_sphere", true)) {
+	String shape = EditorSettings::get_singleton()->get_project_metadata("inspector_options", "material_preview_mesh", "sphere");
+	if (shape == "sphere") {
 		box_instance->hide();
-	} else {
-		box_instance->show();
+		quad_instance->hide();
+	} else if (shape == "box") {
 		sphere_instance->hide();
+		box_instance->show();
+		quad_instance->hide();
+		sphere_switch->set_pressed(false);
 		box_switch->set_pressed(true);
+		quad_switch->set_pressed(false);
+	} else {
+		sphere_instance->hide();
+		box_instance->hide();
+		quad_instance->show();
 		sphere_switch->set_pressed(false);
+		box_switch->set_pressed(false);
+		quad_switch->set_pressed(true);
```

Comment:
Setting the button as pressed emits the pressed signal and runs the same code as above.
I see lots of duplication here, it could be simplified a lot. Also it would be nice to avoid setting project metadata while the editor is being initialized (which I think happens currently, because of the buttons being pressed here).

Reply:
I think I removed all the redundant `set_pressed()` calls. I moved the `connect()` calls to after the few necessary `set_pressed()` calls in `MaterialEditor()`, so project metadata won't be written. Now `MaterialEditor()` should only *read* project metadata, not write it.

Reply:
I forgot to mention that `set_pressed_no_signal()` exists, but connecting later works too.

Reply:
Seems more legible to me to leave the connections with the button setup and use `set_pressed_no_signal()`, so I moved the connections back. 

Should the `set_pressed(false)` in `_on_sphere_switch_pressed`, etc, also be `set_pressed_no_signal()`?

Reply:
It could for consistency, but releasing a button does not emit `pressed` signal.